src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -26,20 +26,14 @@ public static partial class JsonSerializer
             Debug.Assert(state.Current.ReturnValue != default || state.Current.TempDictionaryValues != default);
             Debug.Assert(state.Current.JsonClassInfo != default);
 
-            if ((state.Current.IsProcessingDictionary || state.Current.IsProcessingIDictionaryConstructible) &&
+            if (state.Current.IsProcessingDictionaryOrIDictionaryConstructible() &&
                 state.Current.JsonClassInfo.DataExtensionProperty != state.Current.JsonPropertyInfo)
             {
-                if (state.Current.IsDictionary || state.Current.IsIDictionaryConstructible)
+                if (state.Current.IsProcessingObject(ClassType.Dictionary | ClassType.IDictionaryConstructible))
```

Comment:
Regarding https://github.com/dotnet/corefx/pull/41238#discussion_r326791833, here's a call-site I was referring to, where we are potentially doing redundant work.

`IsProcessingDictionaryOrIDictionaryConstructible` calls `IsProcessingObject` already, and then we call `IsProcessingObject` again with the same flags. Maybe, we can restructure code like this (in addition to using bit flags), to reduce certain checks.

Reply:
Sure that makes sense. I'll create a temp bool to hold the "object" result.

Reply:
Btw, this wasn't the only instance of redundant checks. I was referring to it as an example.